nixos/modules/hardware/corectrl.nix

```
@@ -0,0 +1,62 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.corectrl;
+in
+{
+  options.programs.corectrl = {
+    enable = mkEnableOption ''
+      A tool to overclock amd graphics cards and processors.
+      Add your user to the corectrl group to run corectrl without needing to enter your password
+    '';
+
+    gpuOverclock = {
+      enable = mkEnableOption ''
+        true
+      '';
+      ppfeaturemask = mkOption {
+        type = types.str;
+        default = "0xfffd7fff";
+        example = "0xffffffff";
+        description = ''
+          The overdrive bit for amdgpu
+          Default is 0xfffd7fff as it is less likely to cause flicker issues
+          Set this to 0xfffd7fff TODO
+          Set this to 0xffffffff TODO
```

```suggestion
          Sets the amdgpu.ppfeaturemask kernel option.
          In particular, it is used here to set the overdrive bit.
          Default is 0xfffd7fff as it is less likely to cause flicker issues.
```

I think it should suffice for now.

---

I think 0xffffffff should be mentioned. like this
```
...
If some feature you need isn't enabled, set this to 0xffffffff.
```

---

OK, do it.